spec/controllers/user_sets_controller_spec.rb

```diff
@@ -8,31 +8,76 @@
 
 RSpec.describe UserSetsController do
   describe 'GET show' do
-    let(:set) {
+    let(:set) do
       double(:set, {
         items: [
           double(:record),
           double(:record)
         ]
       })
-    }
+    end
+    let(:set_id) {'1'}
+
     before do
-      allow(Supplejack::UserSet).to receive(:find) {set}
+      expect(Supplejack::UserSet).to receive(:find).with(set_id).and_return(set)
 
-      get :show, id: ''
+      get :show, id: set_id
     end
 
-    it 'assigns @user_set to the set' do
+    it "assigns @user_set to the correct UserSet" do
       expect(assigns(:user_set)).to eq(set)
     end
     
     it 'assigns @records to the sets items' do
-
       expect(assigns(:records)).to eq(set.items)
     end
+  end
+
+  describe 'GET index' do
+    login_user
+
+    it "retrieves the sets from the current_sj_user" do
+      dummy_sj_user = instance_double(Supplejack::User)
+      
+      expect(controller).to receive(:current_sj_user).and_return(dummy_sj_user)
+      expect(dummy_sj_user).to receive(:sets).and_return(
+        [double(name: 'test', id: 1, count: 1)]
+      )
+
+      get :index
```

Comment:
The expects set up expectations in how the objects will be called, the get then triggers the calls. Which is the opposite of a normal test.